gensim/downloader.py

```diff
@@ -0,0 +1,274 @@
+"""This module is an API for downloading, getting information and loading datasets/models."""
+from __future__ import absolute_import
+import argparse
+import os
+import json
+import tarfile
+import logging
+import sys
+import errno
+import hashlib
+import math
+from shutil import rmtree
+import tempfile
+try:
+    import urllib.request as urllib
+except ImportError:
+    import urllib
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+user_dir = os.path.expanduser('~')
+base_dir = os.path.join(user_dir, 'gensim-data')
+logger = logging.getLogger('gensim.api')
+
+
+def progress(chunks_downloaded, chunk_size, total_size):
+    r"""Create and update the progress bar.
+
+    Parameters
+    ----------
+    chunks_downloaded : int
+        Number of chunks of data that have been downloaded
+    chunks_size : int
+        Size of each chunk of data.
+    total_size : int
+        Total size of the dataset/model
+
+    References
+    ----------
+    [1] https://gist.github.com/vladignatyev/06860ec2040cb497f0f3
+
+    """
+    bar_len = 50
+    size_downloaded = float(chunks_downloaded * chunk_size)
+    filled_len = int(math.floor((bar_len * size_downloaded) / total_size))
+    percent_downloaded = round((size_downloaded * 100) / total_size, 1)
+    bar = '=' * filled_len + '-' * (bar_len - filled_len)
+    sys.stdout.write('[%s] %s%s %s/%sMB downloaded\r' % (bar, percent_downloaded, "%", round(size_downloaded / (1024 * 1024), 1), round(float(total_size) / (1024 * 1024), 1)))
+    sys.stdout.flush()
+
+
+def _create_base_dir():
```

Comment:
Maybe use `__` instead of `_` will be better (for hiding from import), here and everywhere?